deepchem/feat/materials_featurizers.py

```diff
@@ -0,0 +1,285 @@
+              (np.matrix(coord_vec) * lattice).A1) * ang_to_bohr
+          sine_mat[i][j] = atomic_numbers[i] * atomic_numbers[j] / \
+                          trig_dist
+        else:
+          sine_mat[i][j] = sine_mat[j][i]
+
+    if self.eig:  # flatten array to eigenvalues
+      eigs, _ = np.linalg.eig(sine_mat)
+      zeros = np.zeros((self.max_atoms,))
+      zeros[:len(eigs)] = eigs
+      eigs = zeros
+      return eigs
+    else:
+      sine_mat = pad_array(sine_mat, self.max_atoms)
+      return sine_mat
+
+
+class StructureGraphFeaturizer(Featurizer):
+  """
+  Calculate structure graph for crystals.
+
+  Create a graph representation of a crystal structure where atoms
+  are nodes and connections between atoms (bonds) are edges. Bonds
+  are determined by choosing a strategy for finding nearest neighbors
+  from pymatgen.analysis.local_env. For periodic
+  graphs, each edge belongs to a lattice image.
+
+  The NetworkX package is used for graph representations.
+  Hagberg, A. et al. SciPy2008, 11-15 (2008).
+
+  This featurizer requires the optional dependency pymatgen. It may
+  be useful when using graph network models and crystal graph
+  convolutional networks.
+
+  #TODO (@ncfrey) process graph features for models
+
+  """
+
+  def __init__(self, strategy=None):
+    """
+    Parameters
+    ----------
+    strategy : pymatgen.analysis.local_env.NearNeighbors
+      An instance of NearNeighbors that determines how graph is constructed.
+
+    """
+
+    if not strategy:
+      from pymatgen.analysis.local_env import MinimumDistanceNN
+      strategy = MinimumDistanceNN()
+
+    self.strategy = strategy
+
+  def _featurize(self, struct):
+    """
+    Calculate structure graph from pymatgen structure.
+
+    Parameters
+    ----------
+    struct : dict
+      Json-serializable dictionary representation of pymatgen.core.structure
+      https://pymatgen.org/pymatgen.core.structure.html
+
+    Returns
+    -------
+    feats: tuple
+      atomic numbers, nodes, and edges in networkx.classes.multidigraph.MultiDiGraph format.
+
+    """
+
+    from pymatgen import Structure
+
+    # Get pymatgen structure object
+    s = Structure.from_dict(struct)
+
+    features = self._get_structure_graph_features(s)
+
+    return features
+
+  def _get_structure_graph_features(self, struct):
+    """
+    Calculate structure graph features from pymatgen structure.
+
+    Parameters
+    ----------
+    struct : pymatgen.core.structure
+      A periodic crystal composed of a lattice and a sequence of atomic
+      sites with 3D coordinates and elements.
+
+    Returns
+    -------
+    feats: tuple
+      atomic numbers, nodes, and edges in networkx.classes.multidigraph.MultiDiGraph format.
+    
+    """
+
+    from pymatgen.analysis.graphs import StructureGraph
+
+    atom_features = np.array([site.specie.Z for site in struct], dtype='int32')
+
+    sg = StructureGraph.with_local_env_strategy(struct, self.strategy)
```

Comment:
From my experience, `with_local_env_strategy ` is too slow... (This issue is discussed in CGCNN reop, see https://github.com/txie-93/cgcnn/pull/18) 
If this featurizer is implemented for GCN or CGCNN, I recommend using `get_all_neighbors ` method like https://github.com/txie-93/cgcnn/blob/d612a69530a72ba686fca56813657b89f2440cc5/cgcnn/data.py#L319-L350

Reply:
Agreed, I think doing something like `get_all_neighbors` is a great idea. I frequently have to transform a bunch of structures into structure graphs and save them and it does take quite awhile.

Reply:
+1 for using `get_all_neighbors` here instead.